### PR TITLE
update OCPDistributionInfo

### PR DESCRIPTION
**Pull request by zhiweiyin318**
story: https://github.com/open-cluster-management/backlog/issues/11698

### CODE REVIEW

File: pkg/apis/internal.open-cluster-management.io/v1beta1/clusterinfo_types.go [hunk 1]
```diff
@@ -29,22 +29,26 @@ type ClusterInfoStatus struct {
 	Version string `json:"version,omitempty"`
 
 	// KubeVendor describes the kubernetes provider of the managed cluster.
+	// Deprecated in release 2.3. Use clusterClaim platform.open-cluster-management.io instead.
```
Comment (qiujian16) [inline pkg/apis/internal.open-cluster-management.io/v1beta1/clusterinfo_types.go:32]: add  a comment that this is to be removed in the future

File: pkg/apis/internal.open-cluster-management.io/v1beta1/clusterinfo_types.go [hunk 2]
```diff
@@ -112,16 +116,84 @@ type DistributionInfo struct {
...
+	// Deprecated in release 2.3. Use clusterClaim version.openshift.io instead.
 	Version string `json:"version,omitempty"`
-	// availableUpdates contains the list of update versions that are appropriate for the manage cluster.
+
+	// AvailableUpdates contains the list of update versions that are appropriate for the manage cluster.
+	// Deprecated in release 2.3. Use VersionAvailableUpdates instead.
 	AvailableUpdates []string `json:"availableUpdates,omitempty"`
-	// Desiredversion represents the desired version of the ocp cluster
+
+	// DesiredVersion is the version that the cluster is reconciling towards
```
Comment (qiujian16) [inline pkg/apis/internal.open-cluster-management.io/v1beta1/clusterinfo_types.go:176]: also deprecate this?

Comment (zhiweiyin318) [reply pkg/apis/internal.open-cluster-management.io/v1beta1/clusterinfo_types.go:176]: need add the whole fileds of [Desired](https://github.com/openshift/api/blob/a3c8fa53e01bc4442b687d3aca1571692f7676ce/config/v1/types_cluster_version.go#L87 )  to the clusterinfo if we deprecate this. I think `DesiredVersion` is enough.

File: pkg/apis/internal.open-cluster-management.io/v1beta1/clusterinfo_types.go (hunk 1 above)
Comment (zhiweiyin318) [reply pkg/apis/internal.open-cluster-management.io/v1beta1/clusterinfo_types.go]: done

File: pkg/klusterlet/clusterinfo/clusterinfo_controller.go [hunk 3]
```diff
@@ -409,36 +409,79 @@ func (r *ClusterInfoReconciler) getOCPDistributionInfo() (clusterv1beta1.OCPDist
...
 	for _, update := range availableUpdates {
-		availableVersion, _, err := unstructured.NestedString(update.(map[string]interface{}), "version")
+		versionUpdate := clusterv1beta1.OCPVersionRelease{}
+		versionUpdate.Version, _, err = unstructured.NestedString(update.(map[string]interface{}), "version")
 		if err != nil {
-			klog.Errorf("failed to get OCP cluster version in latest history of status: %v", err)
-			continue
+			klog.Errorf("failed to get version in availableUpdates of clusterVersion: %v", err)
+		}
+		versionUpdate.Image, _, err = unstructured.NestedString(update.(map[string]interface{}), "image")
```
Comment (qiujian16) [inline pkg/klusterlet/clusterinfo/clusterinfo_controller.go:424]: do we import openshift/api?  if yes, we should render to typed api instead  of using unstructured. Using nested could be erroneous.

Comment (zhiweiyin318) [reply pkg/klusterlet/clusterinfo/clusterinfo_controller.go:424]: no, I tried to import openshift/api, but conflicted with the old version imported in hive.

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**